crates/bevy_pbr/src/ssao/gtao.wgsl

```diff
@@ -0,0 +1,168 @@
+// Ground Truth-based Ambient Occlusion (GTAO)
+// Paper: https://www.activision.com/cdn/research/Practical_Real_Time_Strategies_for_Accurate_Indirect_Occlusion_NEW%20VERSION_COLOR.pdf
+// Presentation: https://blog.selfshadow.com/publications/s2016-shading-course/activision/s2016_pbs_activision_occlusion.pdf
+
+// Source code heavily based on XeGTAO v1.30 from Intel
+// https://github.com/GameTechDev/XeGTAO/blob/0d177ce06bfa642f64d8af4de1197ad1bcb862d4/Source/Rendering/Shaders/XeGTAO.hlsli
+
+#import bevy_pbr::gtao_utils
+#import bevy_pbr::utils
+#import bevy_render::view
+#import bevy_render::globals
+
+@group(0) @binding(0) var preprocessed_depth: texture_2d<f32>;
+@group(0) @binding(1) var normals: texture_2d<f32>;
+@group(0) @binding(2) var hilbert_index: texture_2d<u32>;
+@group(0) @binding(3) var ambient_occlusion: texture_storage_2d<r16float, write>;
+@group(0) @binding(4) var depth_differences: texture_storage_2d<r32uint, write>;
+@group(0) @binding(5) var<uniform> globals: Globals;
+@group(1) @binding(0) var point_clamp_sampler: sampler;
+@group(1) @binding(1) var<uniform> view: View;
+
+fn load_noise(pixel_coordinates: vec2<i32>) -> vec2<f32> {
+    var index = textureLoad(hilbert_index, pixel_coordinates % 64, 0).r;
+
+#ifdef TEMPORAL_NOISE
+    index += 288u * (globals.frame_count % 64u);
+#endif
```

Comment:
The hilbert curve idea is taken from XeGTAO (https://github.com/GameTechDev/XeGTAO#sampling-noise), which is taken from a shadertoy (https://www.shadertoy.com/view/3tB3z3).

The basic idea is to use a hilbert curve to feed into the R2 sequence in order to get multiple random numbers in a stratified way. Iirc, it's been a while since I've looked at this.

288 is just a random number chosen to look nice, it's again copied directly from XeGTAO.

I wouldn't worry too much about this code, it's more or less just copied from XeGTAO. Eventually it will be replaced by spatiotemporal blue noise sampling when I revamp the SSAO denoiser, but I have no plans of doing that any time soon, and this PR is already pretty big as-is.